backend/routers/users.js

```diff
@@ -95,8 +96,47 @@ async function login(req, res, next) {
   }
 }
 
+async function resetUserInfo(req, res, next) {
+  const { username: oldUsername } = req.user;
+  const {
+    username: newUsername, email: newEmail, password: newPassword, kingdomName: newKingdomName,
+  } = req.body;
+  try {
+    let changedTarget = {};
+    if (newUsername) {
+      changedTarget = Object.assign(changedTarget, { username: newUsername });
+    }
+    if (newEmail) {
+      changedTarget = Object.assign(changedTarget, { email: newEmail });
+    }
+    if (newPassword) {
+      changedTarget = Object.assign(changedTarget, { password: newPassword });
+    }
+    if (newKingdomName) {
+      changedTarget = Object.assign(changedTarget, { kingdomName: newKingdomName });
+    }
```

Comment:
can be
`
    const changedTarget = {};
    if (newUsername) {
      changedTarget.username = newUsername;
    }
    if (newEmail) {
      changedTarget.email = newEmail;
    }
    if (newPassword) {
      changedTarget.password = newPassword;
    }
    if (newKingdomName) {
      changedTarget.kingdomName = newKingdomName;
    }
`